src/lambda/commands/deploySamApplication.ts

```diff
@@ -186,35 +204,104 @@ async function buildPackageDeploy(params: {
             params.invoker,
             params.channelLogger.logger,
         )
+    } catch (err) {
```

Comment:
How come only `deployOperation` has a `catch`? This will also generate two logs, one at this level and one from `deploy()` via `outputDeployError`...is this what we want?

Reply:
We don't really want this catch here. If we can move `enhanceAwsCloudFormationInstructions` logic elsewhere (or remote it outright), we could remove this catch.

Reply:
I don't see why we can't call `enhanceAwsCloudFormationInstructions` in the other catch statement, that also has access to `deployParams`.

Unfortunately, we need it if we want to allow the users to do a fast-retry.